### PR TITLE
Fixes error message when _ is used without dev mode

### PR DESCRIPTION
**Pull request by gupnik**
As discussed in #13815

### CODE REVIEW

**Comment by gupnik** [comment]
bot fmt

File: frame/support/procedural/src/pallet/parse/storage.rs [hunk 1]
```diff
@@ -514,10 +514,16 @@ fn process_unnamed_generics(
...
 	let use_default_hasher = |arg_pos| {
-		if let Some(arg) = retrieve_arg(arg_pos).ok() {
-			dev_mode && syn::parse2::<syn::Token![_]>(arg.to_token_stream()).is_ok()
+		let arg = retrieve_arg(arg_pos)?;
+		if syn::parse2::<syn::Token![_]>(arg.to_token_stream()).is_ok() {
+			if dev_mode {
+				Ok(true)
+			} else {
+				let msg = "`_` can only be used in dev_mode. Please specify an appropriate hasher.";
+				Err(syn::Error::new(args_span, msg))
```
Comment (bkchr) [inline frame/support/procedural/src/pallet/parse/storage.rs:523]:
```suggestion
				Err(syn::Error::new(arg.span(), msg))
```

To highlight exactly which argument we mean.

**Review by bkchr — APPROVED** [review]
Ty

File: frame/support/procedural/src/pallet/parse/storage.rs (hunk 1 above)
Comment (gupnik) [reply frame/support/procedural/src/pallet/parse/storage.rs]: Done. Thank you.

**Pull request merged**